app/models/concerns/twitter_user/builder.rb

```diff
@@ -0,0 +1,114 @@
+require 'active_support/concern'
+
+module Concerns::TwitterUser::Builder
+  extend ActiveSupport::Concern
+
+  class_methods do
+    def build_by_user(user, attrs = {})
+      build_relation = attrs.has_key?(:build_relation) ? attrs.delete(:build_relation) : false
+      tu = new(attrs) do |tu|
+        tu.uid = user.id
+        tu.screen_name = user.screen_name
+        tu.user_info = user.slice(*TwitterUser::PROFILE_SAVE_KEYS).to_json # TODO check the type of keys and values
+      end
+      tu.build_relations if build_relation
+      tu
+    end
+
+    def build_by_client(client, user, attrs = {})
+      build_by_user(client.user(user), attrs.merge(client: client))
+    end
+
+    def build(client, user, option = {})
+      build_by_client(client, user, option)
+    end
+
+  end
+
+  included do
+    def build_relations
+      uid_i = uid.to_i
+      search_query = "@#{screen_name}"
+
+      if ego_surfing?
+        candidates = [
+          {method: :friends_parallelly, args: [uid_i]},
+          {method: :followers_parallelly, args: [uid_i]},
+          {method: :user_timeline, args: [uid_i]}, # for replying
+          {method: :search, args: [search_query]}, # for replied
+          {method: :home_timeline, args: [uid_i]},
+          {method: :mentions_timeline, args: [uid_i]}, # for replied
+          {method: :favorites, args: [uid_i]} # for favoriting
+        ]
+        if without_friends
+          candidates = candidates.slice(2, candidates.size - 2)
+          _statuses, _search_results, _, _mentions, _favorites = client._fetch_parallelly(candidates)
```

Comment:
__[rubocop]__
* Do not use prefix `_` for a variable that is used.
* Line is too long. [101/80] [:ref](https://github.com/bbatsov/ruby-style-guide#80-character-limits)